geograypher/meshes/meshes.py

```diff
@@ -162,19 +164,57 @@ def load_mesh(
             simplify_tol_meters=ROI_simplify_tol_meters,
         )
 
-        # Downsample mesh if needed
+        # Downsample mesh if needed transfer textures from original mesh to downsampled mesh
         if downsample_target != 1.0:
             # TODO try decimate_pro and compare quality and runtime
             # TODO see if there's a way to preserve the mesh colors
             # TODO also see this decimation algorithm: https://pyvista.github.io/fast-simplification/
             self.logger.info("Downsampling the mesh")
-            self.pyvista_mesh = self.pyvista_mesh.decimate(
+            intermediate_mesh = self.pyvista_mesh.decimate(
                 target_reduction=(1 - downsample_target)
             )
+            self.pyvista_mesh = self.transfer_texture(intermediate_mesh)
         self.logger.info("Extracting faces from mesh")
         # See here for format: https://github.com/pyvista/pyvista-support/issues/96
         self.faces = self.pyvista_mesh.faces.reshape((-1, 4))[:, 1:4].copy()
 
+    def transfer_texture(self, downsampled_mesh):
+        """Transfer texture from original mesh to a downsampled version using KDTree for nearest neighbor point searches
+
+        Args:
+            downsampled_mesh (pv.PolyData): The downsampled version of the original mesh
+
+        Returns:
+            pv.PolyData: The downsampled mesh with the transferred textures
+        """
+        # Only compute KDTree if active scalars are associated with points
+        if (
+            self.kdtree is None
+            and self.pyvista_mesh.active_scalars_info.association
+            == pv.FieldAssociation.POINT
+        ):
+            # Store original mesh points in KDTree for nearest neighbor search
+            self.kdtree = KDTree(self.pyvista_mesh.points)
+
+            # For ecah point in the downsampled mesh find the nearest neighbor point in the original mesh
+            _, indices = self.kdtree.query(downsampled_mesh.points)
+
+            # Get active scalars name and value at the found indices
+            active_scalars_name = self.pyvista_mesh.active_scalars_name
+            active_scalars = self.pyvista_mesh.active_scalars[indices]
+
+            # Assign active scalars to the downsampled mesh
+            downsampled_mesh[active_scalars_name] = active_scalars
+
+            # Set the active scalars on the downsampled mesh to ensure they are being used
+            downsampled_mesh.set_active_scalars(active_scalars_name)
```

Comment:
As an aside, this might not be strictly necessary, since I think the `[]` notation for setting scalars also updates active scalars to be the most recent one. Feel free to test, but leaving this line is fine to be explicit.

Reply:
The `[]` does update active scalars to be the most recent one. I think it should be fine to remove `downsampled_mesh.set_active_scalars(active_scalars_name)` and I can adjust the comment above to acknowledge that both the`active_scalars` are being assigned and being used.